ddtrace/contrib/starlette/patch.py

```diff
@@ -65,3 +53,37 @@ def unpatch():
     setattr(starlette, "_datadog_patch", False)
 
     _u(starlette.applications.Starlette, "__init__")
+
+    # We need to check that Fastapi instrumentation hasn't already unpatched these
+    if isinstance(starlette.routing.Route.handle, ObjectProxy):
+        _u(starlette.routing.Route, "handle")
+
+    if isinstance(starlette.routing.Mount.handle, ObjectProxy):
+        _u(starlette.routing.Mount, "handle")
+
+
+def traced_handler(wrapped, instance, args, kwargs):
+    # Since handle can be called multiple times for one request, we take the path of each instance
+    # Then combine them at the end to get the correct resource names
+    scope = get_argument_value(args, kwargs, 0, "scope")
+
+    # Add the path to the resource_paths list
+    if "resource_paths" not in scope["datadog"]:
+        scope["datadog"]["resource_paths"] = [instance.path]
+    else:
+        scope["datadog"]["resource_paths"].append(instance.path)
+
+    # Iterate through the request_spans and assign the correct resource name to each
+    if scope["datadog"].get("request_spans"):
+        for index, span in enumerate(scope["datadog"].get("request_spans")):
+            # We want to set the full resource name on the first request span 
+            # And one part less of the full resource name for each proceeding request span
+            # e.g. full path is /subapp/hello/{name}, first request span gets that, second request span gets /hello/{name}
+            path = "".join(scope["datadog"].get("resource_paths")[index:])
```

Comment:
`.get("resource_paths")` can either be `None` or not the same length, we'll want to check for that.